processor/transformprocessor/README.md

```diff
@@ -0,0 +1,57 @@
+# Transform Processor
+
+Supported pipeline types: traces
+
+The transform processor modifies telemetry based on configuration using the Telemetry Query Language.
+It takes a list of queries which are performed in the order specified in the config.
+
+Queries are composed of the following parts
+- Path expressions: Fields within the incoming data can be referenced using expressions composed of the names as defined
+in the OTLP protobuf definition. e.g., `status.code`, `attributes["http.method"]`. If the path expression begins with
+`resource.` or `instrumentation_library.`, it will reference those values.
+  - The name `instrumentation_library` within OpenTelemetry is currently under discussion and may be changed in the future.
+- Literals: Strings, ints, and floats can be referenced as literal values
+- Function invocations: Functions can be invoked with arguments matching the function's expected arguments
+- Where clause: Telemetry to modify can be filtered by appending `where a <op> b`, with `a` and `b` being any of the above.
+
+Supported functions:
+- `set(target, value)` - `target` is a path expression to a telemetry field to set `value` into. `value` is any value type.
+e.g., `set(attributes["http.path"], "/foo")`, `set(name, attributes["http.route"])`. If `value` resolves to `nil`, e.g.
+it references an unset map value, there will be no action.
+
+- `keep(target, string...)` - `target` is a path expression to a map type field. The map will be mutated to only contain
+the fields specified by the list of strings. e.g., `keep(attributes, "http.method")`, `keep(attributes, "http.method", "http.route")`
+
+Supported where operations:
+- `==` - matches telemetry where the values are equal to each other
+- `!=` - matches telemetry where the values are not equal to each other
+
+Example configuration:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+
+exporters:
+  nop
+
+processors:
+  transform:
+    queries:
+      - set(status.code, 1) where attributes["http.path"] == "/health"
+      - keep(resource.attributes, "service.name", "service.namespace", "cloud.region")
```

Comment:
Continuing discussion from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6985#discussion_r777652916

```
Don't remember the final decision about the design (1 processor per signal or 1 processor for all signals). If we have 1 processor for all signals as the name suggest ("transform") then this will apply for any telemetry type, but the first line would apply only to "traces" since the status is a span concept?

I believe would be hard to understand all these subtle things, so maybe we can have a per signal processor and reuse as much code as possible?

Not sure we need the final decision now, but when reading these lines I got confused.
```

_Originally posted by @bogdandrutu in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6985#discussion_r778520439_

Reply:
I see where you're coming from, by blunt I guess I meant that it would be harder to grok IMO to have three separate components that are the same thing conceptually, running functions on query expressions. Would it be better to put the signals into the config?

```
transform:
  spans:
    queries:
  metrics:
    queries:
  all:
    queries:
```

?

Reply:
@bogdandrutu Updated to use per-signal queries in config